src/tribol/mesh/MfemData.cpp

```diff
@@ -471,6 +553,98 @@ void MfemMeshData::SetLORFactor(integer lor_factor)
   );
 }
 
+void MfemMeshData::ComputeElementThicknesses()
+{
+  auto submesh_thickness = std::make_unique<mfem::QuadratureFunction>(
+    new mfem::QuadratureSpace(&submesh_, 0)
+  );
+  submesh_thickness->SetOwnsSpace(true);
+  // All the elements in the submesh are on the contact surface. The algorithm
+  // works as follows:
+  // 1) For each submesh element, find the corresponding parent volume element
+  // 2) Compute the thickness of the parent volume element (det J at element
+  //    centroid)
+  // 3) If no LOR mesh, store this on a quadrature function on the submesh
+  // 4) If there is an LOR mesh, use the CoarseFineTransformation to find the
+  //    LOR elements linked to the HO mesh and store the thickness of the HO
+  //    element on all of its linked LOR elements.
+  for (int submesh_e{0}; submesh_e < submesh_.GetNE(); ++submesh_e)
+  {
+    // Step 1
+    auto parent_bdr_e = submesh_.GetParentElementIDMap()[submesh_e];
+    auto& parent_mesh = const_cast<mfem::ParMesh&>(parent_mesh_);
+    auto& face_el_tr = *parent_mesh.GetBdrFaceTransformations(parent_bdr_e);
+    auto mask = face_el_tr.GetConfigurationMask();
+    auto parent_e = (mask & mfem::FaceElementTransformations::HAVE_ELEM1) ?
+      face_el_tr.Elem1No :
+      face_el_tr.Elem2No;
+    
+    // Step 2
+    // norm = (dx/dxi x dx/deta) / || dx/dxi x dx/deta || on boundary element
+    // centroid
+    auto& parent_fes = *coords_.GetParentGridFn().ParFESpace();
+    mfem::Array<int> be_dofs;
+    parent_fes.GetBdrElementDofs(parent_bdr_e, be_dofs);
+    mfem::DenseMatrix elem_coords(parent_mesh_.Dimension(), be_dofs.Size());
+    for (int d{0}; d < parent_mesh_.Dimension(); ++d)
+    {
+      mfem::Array<int> be_vdofs(be_dofs);
+      parent_fes.DofsToVDofs(d, be_vdofs);
+      mfem::Vector elemvect(be_dofs.Size());
+      coords_.GetParentGridFn().GetSubVector(be_vdofs, elemvect);
+      elem_coords.SetRow(d, elemvect);
+    }
+    auto& be = *parent_fes.GetBE(parent_bdr_e);
```

Comment:
Maybe make note that this gets a single integration point at centroid of element.